### ISSUE TITLE
Error mapping some value object

### ISSUE THREAD

**Issue by EduardoPires** [henkmollema/Dapper-FluentMap#46]
The entire explanation can be found here:
http://stackoverflow.com/questions/40881195/mapping-value-objects-with-dapper

I'm trying to map a value object to my domain model. This nested VO isn't a table is just a VO in my rich domain. Simplified scenario when I get the error at application startup:
```

    public class Cliente
    {
        public Guid Id { get; set; }
        public CPF CPF { get; set; }
    }

    public class CPF
    {
        public string Numero { get; set; }
    }

    public class DapperMappings
    {
        public static void RegisterDapperMappings()
        {
            FluentMapper.Initialize(config =>
            {
                config.AddMap(new ClienteMap());
            });
        }
    }

    public class ClienteMap : EntityMap<Cliente>
    {
        public ClienteMap()
        {
            Map(c => c.CPF.Numero).ToColumn("CPF");
        }
    }

    public class MvcApplication : System.Web.HttpApplication
    {
        protected void Application_Start()
        {
            DapperMappings.RegisterDapperMappings();
        }
    }

```
I always receive the same error at application initialization:

**Index was outside the bounds of the array.**

Exception Details: System.IndexOutOfRangeException: Index was outside the bounds of the array.

Using ASP.NET MVC 5, more details about the Stack Trace can be found at StackOverflow link

**Comment by g-adolph** [comment]
Hey guys actually i use this solution to fix this problem. 
I think this isn't a solution but maybe a good point to start to fix the problem. 
I think fluent mapper must have a more sophisticated reflection algorithms do to this.

[https://gist.github.com/g-adolph/f0eb5b977006b96e12dff27c4e61212f](url)

**Comment by EduardoPires** [comment]
@g-adolph Your gist isn't available :(

**Comment by henkmollema** [comment]
Try copying the link: https://gist.github.com/g-adolph/f0eb5b977006b96e12dff27c4e61212f

**Comment by g-adolph** [comment]
@henkmollema, if I do a pull request fixing this issue will be accept? I fixed this issue and created a few tests.

**Comment by henkmollema** [comment]
@g-adolph I guess I'll take this PR in. If you would like to make one, please do.

**Comment by jbowen014** [comment]
I'm having the same issue. Is there an ETA on a release for this fix?

Thank you.

**Comment by nicolastakashi** [comment]
Some news about it?

**Comment by nicolastakashi** [comment]
Gentlemen!

Based on the discussion we had here in this thread, I took the liberty of looking at the code that @g-adolph put into the gist and implemented it.

I wrote some tests and it seems that everything really works as it should, I did this because talking to several friends of the software community including @EduardoPires we think that this bug is of great importance for the mapping of value objects with domain entities.

I'm waiting for new...
